src/core/server/saved_objects/migrations/model/model.ts

```diff
@@ -362,11 +364,33 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
     const res = resW as ExcludeRetryableEsError<ResponseType<typeof stateP.controlState>>;
 
     if (Either.isRight(res)) {
+      let logs: MigrationLog[] = stateP.logs;
+      let unusedTypesQuery = stateP.unusedTypesQuery;
+      if (isTypeof(res.right, 'unknown_docs_found')) {
+        logs = [
+          ...stateP.logs,
+          { level: 'warning', message: extractIgnoredUnknownDocs(res.right.unknownDocs) },
+        ];
+
+        const unknownTypesQuery = {
+          bool: {
+            must_not: res.right.unknownDocs
+              .map(({ type }) => type)
+              .filter((type, index, types) => types.indexOf(type) === index)
+              .map((type) => ({ term: { type } })),
+          },
+        };
+
+        unusedTypesQuery = addFiltersToQuery(stateP.unusedTypesQuery, unknownTypesQuery);
```

Comment:
Optional: I'm not sure to understand what the `filter` is doing here 😅 . I'm assuming it prevents from adding duplicates?

Would have gone with `[...new Set(res.right.unknownDocs.map(({ type }) => type))].map((type) => ({ term: { type } }))`

Reply:
Correct, the goal of the filter was to eliminate duplicates.
I like your one-liner as well, will update it.